.github/workflows/cancel_runs.yml

```diff
@@ -0,0 +1,17 @@
+name: Cancel redundant builds
+
+on:
+  # pull_request_target runs in the context of the repository,
+  # and does *not* include any code present in the merge request.
+  # We can perform privileged actions like creating comments and
+  # cancelling runs on pull request events.
+  pull_request_target:
+
+jobs:
+  cancel-runs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: styfle/cancel-workflow-action@0.7.0
+        with:
+          workflow_id: 'tests.yml'
+          access_token: ${{ github.token }}
```

Comment:
Do we need this if we remove `push` from `tests.yml`? :thinking:

Reply:
Yep, this cancels any relevant redundant builds from any target and required a project access token to function.